maskit/examples/load_mnist.py

```diff
@@ -0,0 +1,122 @@
+import tensorflow as tf
+import collections
+from sklearn.decomposition import PCA
+from pennylane import numpy as np
+from sklearn.preprocessing import minmax_scale
+
+np.random.seed(42)
+
+MAX_TRAIN_SAMPLES = 11471
+MAX_TEST_SAMPLES = 1952
+
+
+def reduce_image(x):
+    x = np.reshape(x, [1, 28, 28, 1])
+    x = tf.image.resize(x, [4, 4])
+    x = np.reshape(x, [4, 4])
+    return x / 255
+
+
+def remove_contradicting(xs, ys):
+    mapping = collections.defaultdict(set)
+    for x, y in zip(xs, ys):
+        mapping[str(x)].add(y)
+
+    return zip(*((x, y) for x, y in zip(xs, ys) if len(mapping[str(x)]) == 1))
+
+
+def convert_to_binary(x):
+    x_new = []
+    for a in x:
+        c = (a < 0.5).astype(int)
+        c = np.expand_dims(c, axis=0)
+        x_new.append(c)
+
+    x_new = np.concatenate(x_new)
+    print(x_new.shape)
+    return x_new
+
+
+def convert_label(y, classes):
+    # TODO: currently hardcoded to fit with iris dataset
+    # num_classes = len(classes)
+    num_classes = 4
```

Comment:
This looks a bit fishy. I would expect that if you use the fixed ``num_classes``, then assert that ``classes`` has compatible length; otherwise, use the length of ``classes``.

Reply:
Yes, I agree. It was a pretty bad workaround for some issues in the `main.py` and `circuits.py`, but I will just fix them and use the correct numbers.